### DISCUSSION TITLE
Why does pnpm sometimes use softlinks, hardlinks, junctions and sometimes not?

### DISCUSSION THREAD

**Discussion by cowwoc** [pnpm/pnpm#6835] [Q&A] [answered]
When running under Windows, I see some directories are being linked (using Junctions) and in other cases they are not. I was expecting all of them to get linked.

What am I missing?

**Comment by zkochan** [comment, marked as answer]
Files of the packages are hard linked to the content-addressable store from 

```
node_modules/.pnpm/<pkg_id>/node_modules/<pkg_name>
```

Packages are symlinked (using junctions on Windows) into the node_modules directories of the packages that they are dependencies of.
